Objective-C/CBLDatabase.mm

```diff
@@ -184,6 +184,8 @@ - (CBLDatabaseConfiguration*) config {
 
 
 - (CBLDocument*) documentWithID: (NSString*)documentID {
+    CBLAssertNotNil(documentID);
```

Comment:
This isn't necessary either, because the documentID just gets passed, unused, all the way to the CBLDocument initializer, which checks it (CBLDocument.mm:46.)